### PR TITLE
process docs: update for release lead

### PR DESCRIPTION
**Pull request by tpepper**
I've updated a few small things I noted during the beginning of the
cycle.

Signed-off-by: Tim Pepper <tpepper@vmware.com>

### CODE REVIEW

File: release-team/role-handbooks/release-lead.md [hunk 1]
```diff
@@ -68,7 +68,11 @@ A guiding principle for the lead is be an arbiter of decisions, and not the prim
 
 * Feature exceptions are to be reviewed by the owner SIG and brought to the release team for assessment of risk, especially across SIGs
 
-* All issues in the milestone are considered release-blocking until proven otherwise by the owner SIG
+* All issues and PRs in the milestone are considered release-blocking until proven otherwise by the owner SIG.
+    * Issues and PRs are added to the milestone only by members of the [kubernetes-milestone-maintainers" GitHub team]((https://github.com/orgs/kubernetes/teams/kubernetes-milestone-maintainers) which primarily includes SIG leads.
```
Comment (AishSundar) [inline release-team/role-handbooks/release-lead.md]: nit: fix link markdown

File: release-team/role-handbooks/release-lead.md [hunk 2]
```diff
@@ -68,7 +68,11 @@ A guiding principle for the lead is be an arbiter of decisions, and not the prim
...
+    * Typically the release lead will have been granted maintainer priviledgeon this team for adding/removing folks as needing during the release cycle if SIG leads have turned over.  If not and it is needed ask in Slack #sig-contribx to find the current administrator of teams.
```
Comment (AishSundar) [inline release-team/role-handbooks/release-lead.md]: s/priviledgeon/privilege on

File: release-team/role-handbooks/release-lead.md [hunk 3]
```diff
@@ -68,7 +68,11 @@ A guiding principle for the lead is be an arbiter of decisions, and not the prim
...
+    * The release lead may also need to add members of the release team, and should check with and prune prior release team members who are no longer active.
```
Comment (AishSundar) [inline release-team/role-handbooks/release-lead.md]: Can only the release lead add/prune members to the milestone-maintainers list? Also is there a quick overview of how to add members?

File: release-team/role-handbooks/release-lead.md [hunk 4]
```diff
@@ -100,9 +104,10 @@ A guiding principle for the lead is be an arbiter of decisions, and not the prim
   <tr>
     <td>2</td>
     <td><p>- Assist the features lead in collecting planned work from SIGs
-<p>- Schedule weekly release team meetings at 10 am Pacific time on a day that is most acceptable to the team.  These will eventually turn into burndown meetings and occur daily.
+<p>- Schedule weekly release team meetings at 10 am Pacific time on a day that is most acceptable to the team.  These will eventually turn into burndown meetings and occur daily.  Invite the kubernetes-milestone-burndown@googlegroups.com group.  You will need to contact SIG-Release leads to gain access to the SIG's zoom.us account for hosting/recording/posting meeting video.
```
Comment (AishSundar) [inline release-team/role-handbooks/release-lead.md]: thanks for adding more insight into these housekeeping tasks

**Comment by tpepper** [comment]
Additional information pushed in response to @AishSundar 's comments and discussion.

File: release-team/role-handbooks/release-lead.md [hunk 5]
```diff
@@ -90,19 +94,22 @@ A guiding principle for the lead is be an arbiter of decisions, and not the prim
   <tr>
     <td>1</td>
     <td><p>- Start the release cycle
+<p>- Insure you are joined to the following Google Groups: [kubernetes-milestone-burndown](https://groups.google.com/forum/#!forum/kubernetes-milestone-burndown),[kubernetes-sig-release](https://groups.google.com/forum/#!forum/kubernetes-sig-release), [kubernetes-sig-leads](https://groups.google.com/forum/#!forum/kubernetes-sig-leads), [kubernetes-dev](https://groups.google.com/forum/#!forum/kubernetes-dev).
```
Comment (AishSundar) [inline release-team/role-handbooks/release-lead.md]: Not sure why markdown isnt getting properly linked, probably due to lack of space between , and [ ?

Comment (tpepper) [reply release-team/role-handbooks/release-lead.md]: OOooh it's because to make a table there's inline HTML.  Those links should be html not markdown. :\

**Pull request merged**